lib/jyson.js

```diff
@@ -98,33 +101,34 @@ const fillKeys = (json, object, undefinedValue, opts) => {
     }
 
     if (isFunction(json[key])) {
-      const functionResult = json[key]({ object, key, opts });
-      return setValue(jsonResult, key, typeof functionResult === 'undefined' ? undefinedValue : functionResult);
+      const functionResult = json[key]({ object, key, templateOpts, opts });
+      return setValue(jsonResult, key, typeof functionResult === 'undefined' ? opts.undefinedValue : functionResult);
     }
 
     // Nested Object
     if(json[key] !== null && json[key] !== undefined && typeof json[key] === 'object') {
-      return setValue(jsonResult, key, fillKeys(json[key], object, undefinedValue, opts));
+      return setValue(jsonResult, key, fillKeys(json[key], object, templateOpts, opts));
     }
 
     // We should not be able to get here
-    throw `jyson encountered an unknown template value: ${json[key]}`;
+    throw new Error(`jyson encountered an unknown template value: ${json[key]}`);
   });
 
   return jsonResult;
 };
 
-const buildTemplateFunction = (templateObject, buildTemplateFunctionOpts = { undefinedValue: null }) => {
-  const templateFunction = (objects, opts = {}) => {
+const buildTemplateFunction = (templateObject, buildTemplateFunctionOpts = {}) => {
+  const opts = Object.assign({ undefinedValue: null, emptyArrayValue: [] }, buildTemplateFunctionOpts, { arrayIndexes: [] });
```

Comment:
minor: could also spread `buildTemplateFunctionOpts` and `arrayIndexes` here instead of using `Object.assign`, just a bit more readable.

Reply:
sorry, I don't follow, can you explain? Maybe a code example?

Reply:
you can also write it like this:

` { undefinedValue: null, emptyArrayValue: [], ...buildTemplateFunctionOpts, arrayIndexes: [] }`

Reply:
ahh, got it, sadly this is not supported in node 6 and currently we want to support node 6 and up https://github.com/hubba/jyson/blob/master/.travis.yml#L8

Reply:
oh this is on node 6? nvm then